test/metrics/index.js

```diff
@@ -145,6 +145,21 @@ test('reports expected metrics', function (t) {
   metrics.start()
 })
 
+test('applies metrics limit', function (t) {
+  agent = mockAgent({
+    metricsInterval: 10,
+    metricsLimit: 2,
+    hostname: 'foo',
+    environment: 'bar'
+  }, (metricset = {}) => {
+    t.ok(Object.keys(metricset.samples).length <= 2, 'has expected number of metrics')
```

Comment:
I assume this test works on the basis that we by default collect more than two metrics, and hence if we set the limit to 2, we should hit it just by calling `metrics.start()`. But in that case wouldn't we expect the length to be exactly 2?

Reply:
Yes to both things. 😅